FirmataMarshaller.cpp

```diff
@@ -168,6 +168,8 @@ const
   if ( (Stream *)NULL == FirmataStream ) { return; }
   FirmataStream->write(START_SYSEX);
   FirmataStream->write(REPORT_FIRMWARE);
+  FirmataStream->write(firmata::FIRMWARE_MAJOR_VERSION);
```

Comment:
Yes, that's right. However, if someone would like to get the bytes via the parser (theoretically, this would be the primary way), then we need to change the parser to be able to consume the additional bytes that provide the version information and change the protocol to provide them (which it currently does not).

As you pointed out, this could be accomplished by creating `QUERY_FIRMWARE_VERSION` and `QUERY_PROTOCOL_VERSION`, then migrating logic for `REPORT_FIRMWARE` and `REPORT_VERSION` over to them and keep the changes I've proposed to handle the actual reporting of versions and not the query.